### ISSUE TITLE
Shell initialization hooks failing when using devbox global packages with host shell

### ISSUE THREAD

**Issue by rbhanot4739** [jetify-com/devbox#2344]
### What happened?

I installed devbox as per the documentation and then added few global packages. After that when I ran `devbox global list` I was getting the following warning
```
Warning: devbox global is not activated.

Add the following line to your shell's rcfile (e.g., ~/.bashrc or ~/.zshrc)
and restart your shell to fix this:

        eval "$(devbox global shellenv)"
```

So following the docs mentioned [here](https://www.jetify.com/docs/devbox/devbox_global/#add-global-packages-to-your-current-host-shell) I added the shell initialization hooks to to the shell rc file with `eval "$(devbox global shellenv --init-hook)"`, but even then I am getting the following errors:

Bash
```
-bash: export: `BASH_FUNC_pathappend%%=() {  pathremove $1 $2; local PATHVARIABLE=${2:-PATH}; export $PATHVARIABLE="${!PATHVARIABLE:+${!PATHVARIABLE}:}$1"}': not a valid identifier
-bash: export: `BASH_FUNC_pathprepend%%=() {  pathremove $1 $2; local PATHVARIABLE=${2:-PATH}; export $PATHVARIABLE="$1${!PATHVARIABLE:+:${!PATHVARIABLE}}"}': not a valid identifier
-bash: export: `BASH_FUNC_pathremove%%=() {  local IFS=':'; local NEWPATH; local DIR; local PATHVARIABLE=${2:-PATH}; for DIR in ${!PATHVARIABLE}; do if [ "$DIR" != "$1" ]; then NEWPATH=${NEWPATH:+$NEWPATH:}$DIR; fi; done; export $PATHVARIABLE="$NEWPATH"}': not a valid identifier
```

Zsh
```
(eval):export:1: not valid in this context: BASH_FUNC_pathappend%%
/home/rbhanot/.zshrc:140: command not found: fzf
/home/rbhanot/.zshrc:142: command not found: zoxide
```

Even after adding the devbox initialization hook to zshrc/bashrc the globally installed do not seem to be activated and added to path and hence all the tools which I installed via DevBox aren't being loaded(e.g. fzf and zoxide in this case).

### Steps to reproduce

1. Install devbox. 
2. Add some global packages
3. Add the devbox initialization hook to zshrc/bashrc.

### Command

shellenv

### devbox.json

```jsonc
{
  "$schema": "https://raw.githubusercontent.com/jetify-com/devbox/0.13.4/.schema/devbox.schema.json",
  "packages": [
    "neovim@latest",
    "ripgrep@latest",
    "fzf@latest",
    "fd@latest",
    "lazygit@latest",
    "zoxide@latest",
    "eza@latest",
    "delta@latest",
    "bat@latest",
    "tealdeer@latest",
    "tmux@latest",
    "tmuxinator@latest",
    "direnv@latest",
    "jq@latest",
    "yq@latest",
    "sqlite@latest",
    "lua@latest",
    "luajit@latest",
    "luarocks@latest"
  ],
  "shell": {
    "init_hook": [
      "echo 'Welcome to devbox!' > /dev/null"
    ],
    "scripts": {
      "test": [
        "echo \"Error: no test specified\" && exit 1"
      ]
    }
  }
}
```

### Devbox version

0.13.4

### Nix version

nix (Nix) 2.24.7

### What system does this bug occur on?

Linux (x86-64)

**Comment by rbhanot4739** [comment]
@DerArkeN what do you mean by first rc file ?Its already placed at the top in the minimal zshrc i shared above.

**Comment by Lagoja** [comment]
@rbhanot4739 Does this issue persist even after running `refresh-global`? Sometimes that error occurs if you've recently upgraded devbox

**Comment by Lagoja** [comment]
the `BASH_FUNC()` error that you are seeing is a [Shellshock mitigation](https://antonyt.com/blog/2020-03-27/exploiting-cgi-scripts-with-shellshock), because something in either your zshrc or system zshrc is trying to export a function, which is a security vulnerability. From the linked article:

```
Bash no longer generally interprets the x=() {...} environment variable as a function definition. Function definitions in environment variables are now required to be prefixed with BASH_FUNC_. Exporting functions with export -f x will now set a environment variable BASH_FUNC_x%%=() {...} instead of simply x=() {...}.
```

Can you check if there is a `pathappend` or `pathprepend` function exported in your shell rc?

**Comment by rbhanot4739** [comment]
@Lagoja I am using zsh as my primary shell, the bashec example just for test i did in bash. No there is no pathappend or pathprepend in my `.zshrc`. Also this is a fresh installation of devbox, so not sure why `refresh-global` warning is coming in first place

**Comment by Lagoja** [comment]
Which Linux Distro are you using?

**Comment by rbhanot4739** [comment]
I am using azure linux.

**Comment by Lagoja** [comment]
Ah yep, here it is: 

https://github.com/microsoft/azurelinux/blob/3fc3abd10337c6a6db742f5ce79e1fcfb296dfd8/SPECS/zsh/zprofile.rhs#L10

It looks like Azure sets those functions in the zprofile, and Devbox trying to source and reexport them is what's causing the issue

**Comment by rbhanot4739** [comment]
Yes, you are right, so the default shell for the user is being set by ldap profile. So yes initially the default login shell is bash but as a workaround, i do a `exec zsh -l` to in my bashrc to spawn zsh as my login shell.

But thank you so much for taking the time to troubleshoot this issue.

**Comment by Lagoja** [comment]
Can you try adding `export SHELL=/bin/zsh` to your zshrc, and see if that fixes the issue? I can check with the team as well to see if we should modify our detection logic to handle this case.

**Comment by rbhanot4739** [comment]
@Lagoja I am already exporting `SHELL=/bin/zsh` as part of  `exec`ing zsh from `.bashrc` but unfortunately it didn't work. 

Here are the contents of it.
```
if [ -f "/etc/bash.bashrc" ]; then
  source /etc/bash.bashrc
fi
# End ~/.bashrc

if [[ $- == *i* ]]; then
        export SHELL=/bin/zsh
        exec -l /bin/zsh
else
    echo "not an interactive shell"
fi
```
And here is my `.zshrc`
```
eval "$(devbox global shellenv --init-hook)"
source <(fzf --zsh)
eval "$(zoxide init zsh)"
source ~/powerlevel10k/powerlevel10k.zsh-theme
[[ ! -f ~/.p10k.zsh ]] || source ~/.p10k.zsh
```

**Comment by rbhanot4739** [comment]
@Lagoja Exporting `SHELL=/bin/zsh` didn't work, is there anything else I can try ?

Also do I need to run `refresh-global` on every new package install ?
